folio-security/src/main/java/org/folio/security/domain/model/AuthUserPrincipal.java

```diff
@@ -0,0 +1,48 @@
+package org.folio.security.domain.model;
+
+import java.util.UUID;
+import lombok.Data;
+import org.springframework.security.core.AuthenticatedPrincipal;
+
+@Data
+public class AuthUserPrincipal implements AuthenticatedPrincipal {
+
+  private UUID userId;
+  private String authUserId;
+  private String tenant;
+
+  @Override
+  public String getName() {
+    return userId != null ? userId.toString() : null;
+  }
+
+  /**
+   * Sets userId for {@link AuthUserPrincipal} and returns {@link AuthUserPrincipal}.
+   *
+   * @return this {@link AuthUserPrincipal} with new userId value
+   */
+  public AuthUserPrincipal userId(UUID userId) {
```

Comment:
Why do this manually and not use Lombok?

Reply:
Lombok does not have support for with and setters simultaneously